public/app/features/integrated-alerting/components/NotificationChannel/DeleteNotificationChannelModal/DeleteNotificationChannelModal.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FC, useContext, useState } from 'react';
+import { logger } from '@percona/platform-core';
+import { AppEvents } from '@grafana/data';
+import { appEvents } from 'app/core/core';
+import { DeleteModal } from '../../DeleteModal';
+import { NotificationChannelService } from '../NotificationChannel.service';
+import { NotificationChannelProvider } from '../NotificationChannel.provider';
+import { DeleteNotificationChannelModalProps } from './DeleteNotificationChannelModal.types';
+import { Messages } from './DeleteNotificationChannelsModal.messages';
+
+const { title, getDeleteMessage, getDeleteSuccess } = Messages;
+
+export const DeleteNotificationChannelModal: FC<DeleteNotificationChannelModalProps> = ({
+  isVisible,
+  notificationChannel,
+  setVisible,
+}) => {
+  const [pending, setPending] = useState(false);
+  const { summary } = notificationChannel || {};
+  const { getNotificationChannels } = useContext(NotificationChannelProvider);
+  const onDelete = async () => {
+    try {
+      setPending(true);
+      await NotificationChannelService.remove(notificationChannel);
+      setVisible(false);
+      appEvents.emit(AppEvents.alertSuccess, [getDeleteSuccess(summary)]);
+      setPending(false);
```

Comment:
Shouldn't `setPending` and `setVisible` go to the `finally` section?

Reply:
setPending yes should go, but setVisible no because if there's an error we don't want to close the modal. Done!

Reply:
Sure )